azkaban-web-server/src/main/java/azkaban/webapp/servlet/WebUtils.java

```diff
@@ -108,4 +110,27 @@ public static String getRealClientIpAddr(final Map<String, String> httpHeaders,
     return clientIp;
   }
 
+  /**
+   * Gets the actual client IP address inspecting the X-Forwarded-For HTTP header or using the
+   * provided 'remote IP address' from the low level TCP connection from the client.
+   *
+   * If multiple IP addresses are provided in the X-Forwarded-For header then the first one (first
+   * hop) is used
+   *
+   * @param req HttpServletRequest
+   * @return The actual client IP address
+   */
+  public static String getRealClientIpAddr(final HttpServletRequest req) {
```

Comment:
Is it possible to do some refactoring and pass the value of the `X-Forwarded-For` header directly to `getRealClientIpAddr(Map<String, String> httpHeaders, String remoteAddr)` instead of passing a map? Passing a map doesn't seem useful to me.

Reply:
I just refactored the existing logic and put it here from LoginAbstractAzkabanServlet.java 
This is pre-existing and nothing new.
I did find it weird but kept it as is to make sure nothing breaks.

Reply:
I understand you're just reusing existing code but since we're already touching it let's improve it. I don't think you would need to make too many changes.

Reply:
This involves changes to the tests in WebUtilsTest.java which is some work to do them properly. Given the priorities I can follow it up.

Reply:
If we can't address this now, let's then add a TODO in the code mentioning that `getRealClientIpAddr(HttpServletRequest)` will most likely be removed and it shouldn't be used. Also perhaps that `getRealClientIpAddr(Map<String, String>, String)` needs some refactoring.